File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

    void help(TreeNode* root,vector<int> &inorder)
    {
        if(root == nullptr)
        {
            return;
        }
        help(root->left,inorder);
        inorder.push_back(root->val);
        help(root->right,inorder);
    }
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> inorder;
        help(root,inorder);
        return inorder;    
    }
};

File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
       List<Integer> ls = new ArrayList<>();
       Stack<TreeNode> st = new Stack<>();
       TreeNode curr = root;
       while(!st.isEmpty() || curr != null)
       {
             while(curr != null)
             {
                 st.add(curr);
                 curr = curr.left;
             }
             curr = st.peek();
             ls.add(curr.val);
             st.pop();
             curr = curr.right;
       } 
       return ls;
    }
}